prometheus_client/__init__.py

```diff
@@ -434,6 +442,34 @@ def do_GET(self):
         self.wfile.write(generate_latest(REGISTRY))
 
 
+def build_pushgateway_url(job, instance=None, host='localhost', port=9091,
+                          use_fqdn_as_instance=True):
+    '''
+    Build a valid pushgateway url
+    '''
+
+    if instance is None and use_fqdn_as_instance:
+        instance = socket.getfqdn()
+
+    if instance:
+        instancestr = '/instances/{}'.format(instance)
+    else:
+        instancestr = ''
+
+    url = 'http://{}:{}/metrics/jobs/{}{}'.format(host, port, job, instancestr)
+    return url
+
+
+def push_to_gateway(url, registry, timeout=None):
+    '''Push metrics to the given url'''
+
+    if not HAVE_REQUESTS:
+        raise RuntimeError("requests module is not available")
```

Comment:
As painful as it is, we should stick to urllib for the broadest possible compatibility. I want to avoid any dependencies outside of core cpython, so that there's no friction to instrumenting code.